langchain4j-mongodb-atlas/src/test/java/dev/langchain4j/store/embedding/mongodb/MongoDbTestFixture.java

```diff
@@ -0,0 +1,126 @@
+package dev.langchain4j.store.embedding.mongodb;
+
+import com.mongodb.ConnectionString;
+import com.mongodb.MongoClientSettings;
+import com.mongodb.ServerApi;
+import com.mongodb.ServerApiVersion;
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoClients;
+import com.mongodb.client.MongoDatabase;
+import dev.langchain4j.data.segment.TextSegment;
+import dev.langchain4j.model.embedding.EmbeddingModel;
+import dev.langchain4j.model.embedding.onnx.allminilml6v2q.AllMiniLmL6V2QuantizedEmbeddingModel;
+import dev.langchain4j.store.embedding.EmbeddingStore;
+import org.bson.UuidRepresentation;
+import org.junit.jupiter.api.Assumptions;
+import org.testcontainers.mongodb.MongoDBAtlasLocalContainer;
+import org.testcontainers.shaded.com.google.common.collect.Sets;
+
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.function.Function;
+
+import static java.lang.String.format;
+
+class MongoDbTestFixture {
+
+    public static final EmbeddingModel EMBEDDING_MODEL = new AllMiniLmL6V2QuantizedEmbeddingModel();
+    private static final String USERNAME = getenv("MONGODB_ATLAS_USERNAME");
+    private static final String PASSWORD = getenv("MONGODB_ATLAS_PASSWORD");
+    private static final String HOST = getenv("MONGODB_ATLAS_HOST");
+    private static final String CONNECTION_STRING = getenv("MONGODB_CONNECTION_STRING");
+
+    private static final String DATABASE_NAME = "test_database";
+    private static MongoDBAtlasLocalContainer mongodb;
+
+    private final String collectionName;
```

Comment:
I think`collectionName` can be removed and MongoDbTestFixture turned to singleton.

Reply:
It is necessary to track the name, so that the collection can be dropped after the test. This is in line with what you suggested 1 month ago in [this](https://github.com/langchain4j/langchain4j/pull/1932#discussion_r1802868506) comment:

> Let's maybe create unique tables for each test. 

The search indexes are eventually consistent in MongoDB, and ensuring that we use a new collection each time prevents possible conflicts or test errors.